File: backend/LogisticsAssetTracker.Api/Domain/Entities/Asset.cs
using LogisticsAssetTracker.Api.Domain.Enums;

namespace LogisticsAssetTracker.Api.Domain.Entities;

public class Asset
{
    public Guid Id { get; set; }
    public string AssetCode { get; set; } = string.Empty;
    public string NormalizedAssetCode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public AssetType Type { get; set; }
    public Guid CurrentLocationId { get; set; }
    public Location? CurrentLocation { get; set; }
    public AssetStatus Status { get; set; }
    public AssetCondition Condition { get; set; }
    public Guid? AssignedToUserId { get; set; }
    public User? AssignedToUser { get; set; }
    public string QrCodeValue { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/asset.ts
export type AssetType =
  | "Pallet"
  | "Crate"
  | "Box"
  | "Container"
  | "Tool"
  | "Device"
  | "WarehouseEquipment";

export type AssetStatus = "Available" | "InTransit" | "Delivered" | "Lost" | "Retired";

export type AssetCondition = "Good" | "Damaged" | "NeedsInspection";

export interface Asset {
  id: string;
  assetCode: string;
  name: string;
  type: AssetType;
  currentLocationId: string;
  currentLocationName: string;
  status: AssetStatus;
  condition: AssetCondition;
  assignedToUserId?: string | null;
  assignedToUserName?: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAssetRequest {
  name: string;
  type: AssetType;
  currentLocationId: string;
  status: AssetStatus;
  condition: AssetCondition;
  assignedToUserId?: string;
}

export interface UpdateAssetRequest {
  name: string;
  type: AssetType;
  assignedToUserId?: string;
}

export interface AssetListQuery {
  status?: AssetStatus;
  condition?: AssetCondition;
  locationId?: string;
  type?: AssetType;
  search?: string;
  isActive?: boolean;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
}

export interface AssetQr {
  assetId: string;
  assetCode: string;
  qrCodeValue: string;
  qrUrl: string;
}
